convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,273 @@
+        from Oracle Linux, the redhat/ directory is already used.
+        """
+        super(MoveGrubFiles, self).run()
+
+        if not grub.is_efi():
+            logger.info("Unable to collect data about UEFI on a BIOS system, did not perform moving of Grub2 files.")
+            return
+
+        if systeminfo.system_info.id != "centos":
+            logger.debug("Did not perform moving of GRUB files - only related to CentOS Linux.")
+            return
+
+        # TODO(pstodulk): check behaviour for efibin from a different dir or with a different name for the possibility of
+        #  the different grub content...
+        # E.g. if the efibin is located in a different directory, are these two files valid?
+        logger.info("Moving GRUB2 configuration files to the new UEFI directory %s." % RHEL_EFIDIR_CANONICAL_PATH)
+        src_files = [
+            os.path.join(CENTOS_EFIDIR_CANONICAL_PATH, filename) for filename in ["grubenv", "grub.cfg", "user.cfg"]
+        ]
+        required = src_files[:2]
+
+        # If at least one file exists, this will be skipped. Otherwise, if all
+        # are missing, this will be a hit.
+        if not any(os.path.exists(filename) for filename in src_files):
+            # Get a list of files that are missing that are required and does
+            # not exist.
+            missing_files = [
+                filename for filename in src_files if filename in required and not os.path.exists(filename)
+            ]
+            # without the required files user should not reboot the system
+            self.set_result(
+                level="ERROR",
+                id="UNABLE_TO_FIND_REQUIRED_FILE_FOR_GRUB_CONFIG",
+                title="Couldn't find system GRUB config",
+                description="Couldn't find one of the GRUB config files in the current system which is required for configuring UEFI for RHEL: {}".format(
+                    ", ".join(missing_files)
+                ),
+            )
+            return
+
+        for src_file in src_files:
+            # Skip non-existing file in destination directory
+            if not os.path.exists(src_file):
+                logger.debug(
+                    "The %s file does not exist in %s folder. Moving skipped."
+                    % (os.path.basename(src_file), CENTOS_EFIDIR_CANONICAL_PATH)
+                )
+                continue
+            # Skip already existing file in destination directory
+            dst_file = os.path.join(RHEL_EFIDIR_CANONICAL_PATH, os.path.basename(src_file))
+            if os.path.exists(dst_file):
+                logger.debug(
+                    "The %s file already exists in %s folder. Moving skipped."
+                    % (os.path.basename(src_file), RHEL_EFIDIR_CANONICAL_PATH)
+                )
+                continue
+
+            logger.info("Moving '%s' to '%s'" % (src_file, dst_file))
+
+            try:
+                shutil.move(src_file, dst_file)
```

Comment:
I'm thinking if we shouldn't go back to the `copy2`, since it doesn't make a big difference (the `/boot/efi/EFI/centos` won't be removed) and as we discussed here https://github.com/oamg/convert2rhel/pull/1256#issuecomment-2291248107 we can refactor this later and for now we should be closer to the original code.

If we decide to move, sorry Preston for an extra work you had with my ideas

Reply:
Yes we're sticking with `move`, it's a good addition